examples/todomvc/src/cljs/todomvc/events.cljs

```diff
@@ -21,8 +22,26 @@
 (re-posh/reg-event-ds
  :create-todo-form/create-todo
  (fn [_ [_ id value]]
-   [[ :db/add id :create-todo-form/title "" ]
+   [[ :db/add id :create-todo-form/title ""]
     { :db/id       -1
       :app/type    :type/task
       :task/title  value
       :task/done?  false}]))
+
+;; Retrieval of entries from the Server
+(re-frame/reg-event-fx
+  :retrieve
+  [(re-frame/inject-cofx :ds)] ;; inject coeffect
+  (fn [{:keys [ds]} [_ signal params]] ;; ds here is the DataScript database
+    (GET
+      "/upload"
+      :format :edn
+      :handler #(re-frame/dispatch  [:add-entries  %1])
+      :error-handler #(js/alert (str "Retrieval of entries failed!")))
+    ds))
```

Comment:
In this particular case, you need nothing from outside world in this event. So, no need for coeffect injecting and function signature can be much shorter. Here the adjusted variant:
```clojure
(re-frame/reg-event-fx
  :retrieve
  (fn [] 
    (GET
      "/upload"
      :format :edn
      :handler #(re-frame/dispatch  [:add-entries  %1])
      :error-handler #(js/alert (str "Retrieval of entries failed!")))
    {}))
```
With this everything works just fine. I would also recomment using [effect](https://github.com/Day8/re-frame/blob/master/docs/Effects.md) for sending events, not the pure event (as far as you are not going to do anything with database)